engine/src/render/cascaded_shadow_maps/plugin.cpp

```diff
@@ -0,0 +1,82 @@
+#include <cubos/core/io/window.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+
+#include <cubos/engine/render/camera/camera.hpp>
+#include <cubos/engine/render/camera/plugin.hpp>
+#include <cubos/engine/render/cascaded_shadow_maps/plugin.hpp>
+#include <cubos/engine/render/shadows/directional_caster.hpp>
+#include <cubos/engine/render/shadows/plugin.hpp>
+#include <cubos/engine/window/plugin.hpp>
+
+using cubos::core::io::Window;
+
+CUBOS_DEFINE_TAG(cubos::engine::createCascadedShadowMapsTag);
+CUBOS_DEFINE_TAG(cubos::engine::drawToCascadedShadowMapsTag);
+
+void cubos::engine::cascadedShadowMapsPlugin(Cubos& cubos)
+{
+    cubos.depends(cameraPlugin);
+    cubos.depends(shadowsPlugin);
+    cubos.depends(windowPlugin);
+
+    cubos.tag(createCascadedShadowMapsTag);
+    cubos.tag(drawToCascadedShadowMapsTag).after(createCascadedShadowMapsTag);
+
+    cubos.system("create cascaded shadow maps")
+        .tagged(createCascadedShadowMapsTag)
+        .call([](const Window& window, Query<DirectionalShadowCaster&> query, Query<Entity, const Camera&> cameras) {
+            for (auto [caster] : query)
+            {
+                // Remove shadow maps for cameras that no longer exist
+                std::vector<Entity> removedCameras;
+                for (auto& [cameraEntity, shadowMap] : caster.shadowMaps)
+                {
+                    if (!cameras.at(cameraEntity))
+                    {
+                        removedCameras.push_back(cameraEntity);
+                    }
```

Comment:
Maybe here we should also discard inactive cameras. What do you think?

Reply:
Sounds good!